libs/opengl/src/CCylinder.cpp

```diff
@@ -59,7 +59,42 @@ void CCylinder::render_dl() const
 
 #endif
 }
-
+void CCylinder::serializeTo(mrpt::serialization::CSchemeArchiveBase& out) const
+{
+	out["datatype"] = std::string(this->GetRuntimeClass()->className);
+	out["version"] = 1;
+	out["baseRadius"] = mBaseRadius;
+	out["topRadius"] = mTopRadius;
+	out["height"] = mHeight;
+	out["slices"] = mSlices;
+	out["stacks"] = mStacks;
+	out["hasBottomBase"] = mHasBottomBase;
+	out["hasTopBase"] = mHasTopBase;
+}
+void CCylinder::serializeFrom(mrpt::serialization::CSchemeArchiveBase& in)
+{
+	uint8_t version = static_cast<int>(in["version"]);	// default is 0
+	if(static_cast<std::string>(in["datatype"]) ==
+		std::string(this->GetRuntimeClass()->className))	// match the class name
```

Comment:
idem here: I think it would be better to define a macro that contains code like this if(), but which, if the class name does not match, it raises an exception explaining the error. As it is now, the error would pass unnoticed (!), right?